marathon/marathon_stub.go

```diff
@@ -4,46 +4,56 @@ import (
 	"errors"
 
 	"github.com/allegro/marathon-consul/apps"
+	"sync"
 )
 
 type MarathonerStub struct {
-	AppsStub     []*apps.App
-	AppStub      map[apps.AppID]*apps.App
-	TasksStub    map[apps.AppID][]*apps.Task
-	leader       string
-	interactions bool
+	AppsStub       []*apps.App
+	AppStub        map[apps.AppID]*apps.App
+	TasksStub      map[apps.AppID][]*apps.Task
+	leader         string
+	interactionsMu sync.RWMutex
+	interactions   bool
```

Comment:
How about changing interactions to int and count them. Then we can use [atomic](https://golang.org/pkg/sync/atomic/) functions on it and we do not need mutex.